R/plot_taxa.R

```diff
@@ -13,14 +15,58 @@
 #' @export
 #' 
 #' @examples
-#' observation <- ants_L1[[1]]$tables$observation
-#' id <- names(ants_L1)
+#' \dontrun{
+#' # plot ecocomDP dataset
+#' plot_taxa_accum_sites(
+#'   dataset = ants_L1)
 #' 
-#' plot_taxa_accum_sites(observation, id)
+#' # plot flattened ecocomDP data
+#' plot_taxa_accum_sites(
+#'   flat_data =  flatten_dataset(ants_L1))
 #' 
-plot_taxa_accum_sites <- function(observation, id, alpha = 1) {
+#' # plot an ecocomDP observation table
+#' plot_taxa_accum_sites(
+#'   observation = ants_L1[[1]]$tables$observation)
+#' 
+#' # tidy syntax
+#' ants_L1 %>% plot_taxa_accum_sites()
+#' 
+#' # tidy syntax, filter data by date
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     lubridate::as_date(datetime) > "2003-07-01") %>%
+#'   plot_taxa_accum_sites(flat_data = .)
+#' }
+plot_taxa_accum_sites <- function(
```

Comment:
These changes offer 2 options for getting the same result, and automatically getting the data set ID when it's available. 

It seems that the rationale for accommodating multiple inputs is to fetch the observation table columns from an input table regardless of its specific format. If this is the case, I recommend updating the observation argument description to read `#' @param observation (tbl_df, tbl, data.frame) The observation table, or any table containing containing the equivalent columns.` and not adding any new arguments.

Reply:
that makes sense. I still think we should have an option for allowing the user to pass along the full dataset, so I propose we have the arguments "observation" and "dataset"

Reply:
Another approach for consideration is to subsume `observation`, `dataset`, and `flat_data` arguments under one `data` argument, thereby allowing the first position of this function to be used by the input data type, rather than having to always specify the target argument (i.e. would allow `plot_taxa_accum_sites(data)` where `data` could be `observation`, `dataset`, or `flat_data`). This would also simplify backwards compatibility with the current release, since the common practice is to use the first position without referencing the argument (i.e. `plot_taxa_accum_sites(obs)` rather than `plot_taxa_accum_sites(observation = obs)`. Of course this would require some type of detection on part of the code base, but I think it wouldn't amount to much more than calling the type validation functions on the function inputs (these are proposed above in `flatten_data()` comments).

Reply:
@clnsmth I'm game for the one argument for data approach. I had code that could detect the format, I can put it back in. Can we also implement this approach for flatten_data?

Reply:
Sounds good @sokole. Makes sense to use in `flatten_data()`.

Reply:
@sokole Add deprecation notice and handling of input when the observation parameter is assigned explicitly (e.g. `plot_taxa_accum_sites(observation = data)`). See: https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/Deprecated